[BUG] SideMenu 타입 정의 충돌 - children 속성 타입 에러

## 버그 설명

`SideMenuProps` 인터페이스가 `CollapsiblePrimitive.Root`를 extends할 때 `children` 타입이 충돌하여 TypeScript 컴파일 에러가 발생했습니다.

## 재현 방법

1. `packages/ui/src/organisms/side-menu/side-menu.types.ts` 파일 확인
2. `pnpm --filter a-iot type-check` 실행
3. 다음 타입 에러 발생:

```
error TS2430: Interface 'SideMenuProps' incorrectly extends interface 'Omit<CollapsibleProps & RefAttributes<HTMLDivElement>, "ref">'.
  Types of property 'children' are incompatible.
    Type 'ReactNode | (({ open }: { open: boolean; }) => ReactNode)' is not assignable to type 'ReactNode'.
```

## 예상 동작

- `SideMenuProps`가 함수형 children을 지원하면서도 타입 에러가 발생하지 않아야 함
- TypeScript 컴파일이 성공해야 함

## 실제 동작

- `CollapsiblePrimitive.Root`의 `children: ReactNode` 타입과
- `SideMenuProps`의 `children: ReactNode | (({ open }) => ReactNode)` 타입이 충돌
- 빌드 실패

## 환경

- **OS**: Windows 11
- **Node Version**: >=22.0.0
- **TypeScript Version**: 5.x
- **Package**: @plug-atlas/ui

## 해결 방법

`SideMenuProps`에서 `children` 속성을 `Omit`으로 제외한 후 재정의:

```typescript
// 수정 전
export interface SideMenuProps extends React.ComponentPropsWithoutRef<typeof CollapsiblePrimitive.Root> {
  children?: React.ReactNode | (({ open }: { open: boolean }) => React.ReactNode)
}

// 수정 후
export interface SideMenuProps extends Omit<React.ComponentPropsWithoutRef<typeof CollapsiblePrimitive.Root>, 'children'> {
  children?: React.ReactNode | (({ open }: { open: boolean }) => React.ReactNode)
}
```

## 영향 범위

- `packages/ui/src/organisms/side-menu/` 컴포넌트
- `apps/a-iot` 빌드 프로세스

## 추가 정보

이 이슈는 Radix UI의 `CollapsiblePrimitive.Root`와 커스텀 함수형 children 패턴을 함께 사용할 때 발생하는 일반적인 타입 충돌 케이스입니다.

--- a/packages/ui/src/organisms/side-menu/side-menu.types.ts
+++ b/packages/ui/src/organisms/side-menu/side-menu.types.ts
@@ -1,7 +1,7 @@
 import * as React from "react"
 import * as CollapsiblePrimitive from "@radix-ui/react-collapsible"
 
-export interface SideMenuProps extends React.ComponentPropsWithoutRef<typeof CollapsiblePrimitive.Root> {
+export interface SideMenuProps extends Omit<React.ComponentPropsWithoutRef<typeof CollapsiblePrimitive.Root>, 'children'> {
   defaultOpen?: boolean
   collapsible?: boolean
   children?: React.ReactNode | (({ open }: { open: boolean }) => React.ReactNode)

